component/component_health.go

```diff
@@ -96,3 +96,37 @@ func (ht *HealthType) UnmarshalText(text []byte) error {
 	}
 	return nil
 }
+
+// LeastHealthy returns the Health from the provided arguments which is
+// considered to be the least healthy.
+//
+// Health types are first prioritized by [HealthTypeExited], followed by
+// [HealthTypeUnhealthy], [HealthTypeHealthy], and [HealthTypeUnknown].
```

Comment:
I'm not sure if this order makes sense; it's a bit weird for `HealthTypeHealthy` to take precedence over `HealthTypeUnknown`. Should I flip the order of these two?

Reply:
Ya I would flip. If one of the statuses is unknown I think we could argue the overall status is unknown

Reply:
Ok, flipped 👍